### PR TITLE
Datumizing the overmap.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Converts the overmap from a hardcoded single level to a datum system, allowing for multiple distinct overmaps to exist at once.

## Why and what will this PR improve
This will (in the future!) allow for non-space overmaps; something like landing on a planet from the overmap and then driving a buggy around on a second planetary overmap.

## Authorship
Myself.

## Changelog
Initially, nothing player facing.

### CODE REVIEW

File: code/game/machinery/hologram.dm [hunk 1]
```diff
@@ -56,6 +56,7 @@ var/global/const/HOLOPAD_MODE = RANGE_BASED
 	var/base_icon = "holopad-B"
 
 	var/allow_ai = TRUE
+	var/static/list/reachable_overmaps = list(OVERMAP_ID_SPACE)
```
Comment (comma) [inline code/game/machinery/hologram.dm:59]: maybe should just grab whatever overmap is our location is on?
also might need third type for super long range for planet-space comms

Comment (MistakeNot4892) [reply code/game/machinery/hologram.dm:59]: Thing is, a single z-level can be in multiple overmaps (exoplanet has an exoplanet loc and presumably a loc on its own overmap) so how do you tell which one to use?

Comment (MistakeNot4892) [reply code/game/machinery/hologram.dm:59]: This is no longer relevant since I gave up on multiple overmap entities per site.

Comment (MistakeNot4892) [reply code/game/machinery/hologram.dm:59]: I'm going to stick with the list version just so it's predictable what overmap the holopad reaches.

**Comment by MistakeNot4892** [comment]
I think this is now ready to go. Just needs some more testing. I can join, get to the overmap, and pilot the ship into an electrical storm.

**Comment by comma** [comment]
```
## UNIT_TEST ##: [AREA: Areas shall be pure]: Deep Space (/area/overmap) ([0x400001e]) (1,1,5) has an edited name.
## UNIT_TEST ##: !!! FAILURE !!! [AREA: Areas shall be pure]: Found 1 impure area.
```

File: code/modules/overmap/_overmap.dm [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	..()
+
+/datum/overmap/proc/generate_overmap()
+	testing("Building overmap [name]...")
+	INCREMENT_WORLD_Z_SIZE
+	assigned_z = world.maxz
+	testing("Putting [name] on [assigned_z].")
+
+	var/area/overmap/A = new map_area_type
+	A.name = name
```
Comment (comma) [inline code/modules/overmap/_overmap.dm]: need to add it to area purity test exemptions

Comment (MistakeNot4892) [reply code/modules/overmap/_overmap.dm]: Why was this not failing before?

Comment (comma) [reply code/modules/overmap/_overmap.dm]: Because it wasn't a thing before, it's new code? Overmap area just used default name

Comment (MistakeNot4892) [reply code/modules/overmap/_overmap.dm]: Removed the name setting, overmap datums can just define their own areas properly.

**Review by comma — APPROVED** [review]

**Pull request merged**